### PR TITLE
Mindy/biggerbuttons

### LINKED ISSUE #195: make buttons bigger EVERYWHERE

### PR DESCRIPTION
**Pull request by mindylou**
- increase button tap radius with new UIButton subclass `UtilityButton` (i don't like this name, new name ideas are welcome) 
- fix player slider issue with mini player where the progress started partway into the view
- closes #195

### CODE REVIEW

File: Podcast/MiniPlayerView.swift [hunk 1]
```diff
@@ -52,15 +52,16 @@ class MiniPlayerView: UIView {
...
+        miniPlayerSlider.setThumbImage(UIImage(), for: .normal)
         addSubview(miniPlayerSlider)
         miniPlayerSlider.snp.makeConstraints { make in
             make.top.equalToSuperview()
             make.width.equalToSuperview()
             make.height.equalTo(miniPlayerSliderHeight)
         }
         
-        arrowButton = UIButton(frame: .zero)
+        arrowButton = UtilityButton(frame: .zero)
```
Comment (natashaarmbrust) [inline Podcast/MiniPlayerView.swift]: can we just do UtilityButton() for all of these and make an init in UtilityButton that calls super with a frame of .zero

File: Podcast/PlayerEpisodeDetailView.swift [hunk 2]
```diff
@@ -77,7 +77,7 @@ class PlayerEpisodeDetailView: UIView {
         descriptionTextView.backgroundColor = .clear
         addSubview(descriptionTextView)
         
-        seeMoreButton = UIButton(frame: CGRect(x: 0, y: 0, width: seeMoreButtonWidth, height: seeMoreButtonHeight))
+        seeMoreButton = UtilityButton(frame: CGRect(x: 0, y: 0, width: seeMoreButtonWidth, height: seeMoreButtonHeight))
```
Comment (natashaarmbrust) [inline Podcast/PlayerEpisodeDetailView.swift]: yea I would prefer
` seeMoreButton  = UtilityButton();
seeMoreButton.frame = ...`

File: Podcast/UtilityButton.swift [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+//
+
+import UIKit
+
+class UtilityButton: UIButton {
+    // Minimum touch targets for interactive elements per Apple human interface guidelines
+    let tapWidth: CGFloat = 44
+    let tapHeight: CGFloat = 44
+
+    override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
```
Comment (natashaarmbrust) [inline Podcast/UtilityButton.swift]: nice! would not have thought of this solution!

File: Podcast/UtilityButton.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+//  UtilityButton.swift
+//  Podcast
+//
+//  Created by Mindy Lou on 11/8/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+class UtilityButton: UIButton {
```
Comment (natashaarmbrust) [inline Podcast/UtilityButton.swift]: maybe just `Button` instead of `UtilityButton`

File: Podcast/UtilityButton.swift (hunk 3 above)
Comment (natashaarmbrust) [reply Podcast/UtilityButton.swift]: Will this cause buttons to conflict?

File: Podcast/UtilityButton.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    // Minimum touch targets for interactive elements per Apple human interface guidelines
```
Comment (natashaarmbrust) [inline Podcast/UtilityButton.swift]: love the research

File: Podcast/MiniPlayerView.swift [hunk 6]
```diff
@@ -10,10 +10,10 @@ class MiniPlayerView: UIView {
     
     let miniPlayerHeight: CGFloat = 60.5
     let marginSpacing: CGFloat = 17
-    let buttonSize: CGSize = CGSize(width: 15, height: 18)
+    let buttonSize: CGSize = CGSize(width: 20, height: 24)
```
Comment (natashaarmbrust) [inline Podcast/MiniPlayerView.swift]: this actually looks a bit wonky on my phone in terms of size ... make it a tiny bit smaller

File: Podcast/MiniPlayerView.swift [hunk 7]
```diff
@@ -52,15 +52,16 @@ class MiniPlayerView: UIView {
...
         addSubview(miniPlayerSlider)
         miniPlayerSlider.snp.makeConstraints { make in
             make.top.equalToSuperview()
             make.width.equalToSuperview()
             make.height.equalTo(miniPlayerSliderHeight)
         }
         
-        arrowButton = UIButton(frame: .zero)
+        arrowButton = UtilityButton(frame: .zero)
         arrowButton.setBackgroundImage(#imageLiteral(resourceName: "backArrowDown"), for: .normal)
```
Comment (natashaarmbrust) [inline Podcast/MiniPlayerView.swift]: can you message Eileen for a darker backArrowDown and backArrowUp icon and replace these?
Also message her for the transparent play and pause button backgrounds and include that in this PR!

**Review by natashaarmbrust — COMMENTED** [review]
Looks good! A couple comments

File: Podcast/MiniPlayerView.swift (hunk 7 above)
Comment (mindylou) [reply Podcast/MiniPlayerView.swift]: eileen's having trouble exporting the play/pause with transparent backgrounds for some reason so i'm gonna leave them in for this pr

File: Podcast/UtilityButton.swift (hunk 3 above)
Comment (mindylou) [reply Podcast/UtilityButton.swift]: i tested it and i don't think it does, if two buttons are super close together i think the button that will be tapped is the one added the view last

**Pull request merged**